src/Features/CSharp/Portable/Completion/CompletionProviders/InternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal sealed class InternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectIdKey = "ProjectId";
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            if (ch == '\"')
+            {
+                return true;
+            }
+            return base.IsInsertionTrigger(text, insertedCharacterPosition, options);
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var syntaxTree = await context.Document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            if (syntaxTree.IsEntirelyWithinStringLiteral(context.Position, cancellationToken))
+            {
+                var token = syntaxTree.FindTokenOnLeftOfPosition(context.Position, cancellationToken);
+                var attr = GetAttributeSyntaxOfToken(token);
+                if (attr == null)
+                {
+                    return;
+                }
+                if (attr.Name.TryGetNameParts(out IList<string> nameParts) && nameParts.Count > 0)
+                {
+                    var lastName = nameParts[nameParts.Count - 1];
+                    if (lastName == "InternalsVisibleTo" || lastName == "InternalsVisibleToAttribute")
+                    {
+                        AddAssemblyCompletionItems(context, cancellationToken);
+                    }
+                }
+            }
+        }
+
+        private static AttributeSyntax GetAttributeSyntaxOfToken(SyntaxToken token)
+        {
+            //Supported cases:
+            //[Attribute("|
+            //[Attribute(parameterName:"Text|")
+            //Also supported but excluded by syntaxTree.IsEntirelyWithinStringLiteral in ProvideCompletionsAsync
+            //[Attribute(""|
+            //[Attribute("Text"|)
+            var node = token.Parent;
+            if (node is LiteralExpressionSyntax && node.Kind() == SyntaxKind.StringLiteralExpression)
+            {
+                node = node.Parent;
+                if (node is AttributeArgumentSyntax)
+                {
+                    node = node.Parent;
+                    if (node is AttributeArgumentListSyntax)
```

Comment:
check not needed.  All attribute arguments are parented by AttributeArgumentLists.